csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/InferenceTest.cs

```diff
@@ -157,32 +167,22 @@ public void TestSessionOptions()
 #if USE_TENSORRT
                 opt.AppendExecutionProvider_Tensorrt(0);
 #endif
-#if USE_XNNPACK
-                opt.AppendExecutionProvider("XNNPACK");
-#else
-                ex = Assert.Throws<OnnxRuntimeException>(() => { opt.AppendExecutionProvider("XNNPACK"); });
-                Assert.Contains("XNNPACK execution provider is not supported in this build", ex.Message);
-#endif
-#if USE_SNPE
-                opt.AppendExecutionProvider("SNPE");
-#else
-                ex = Assert.Throws<OnnxRuntimeException>(() => { opt.AppendExecutionProvider("SNPE"); });
-                Assert.Contains("SNPE execution provider is not supported in this build", ex.Message);
-#endif
-#if USE_QNN
-                opt.AppendExecutionProvider("QNN");
-#else
-                ex = Assert.Throws<OnnxRuntimeException>(() => { opt.AppendExecutionProvider("QNN"); });
-                Assert.Contains("QNN execution provider is not supported in this build", ex.Message);
-#endif
-#if USE_COREML
-                opt.AppendExecutionProvider("CoreML");
-#else
-                ex = Assert.Throws<OnnxRuntimeException>(() => { opt.AppendExecutionProvider("CoreML"); });
-                Assert.Contains("CoreML execution provider is not supported in this build", ex.Message);
-#endif
-
-                opt.AppendExecutionProvider_CPU(1);
+                AssertUtils.IfThrowsCheckException<OnnxRuntimeException>(
```

Comment:
now I understand that these `opt.AppendExecutionProvider()` calls may be expected to throw if we don't know whether the EP was enabled in the build at the time of compiling this file. it may be good to add a comment about this.

however, why are the preceding exception checks also possibly expecting success? e.g., for setting an invalid graph optimization level or adding a session config entry with an empty key.

Reply:
Oops, good catch -- replaced with a method that will fail if no exception is thrown